community/logging/src/main/java/org/neo4j/logging/FormattedLog.java

```diff
@@ -41,23 +40,21 @@
  */
 public class FormattedLog extends AbstractLog
 {
-    static final Supplier<Date> DEFAULT_CURRENT_DATE_SUPPLIER = Date::new;
     static final Function<OutputStream, PrintWriter> OUTPUT_STREAM_CONVERTER =
             outputStream -> new PrintWriter( new OutputStreamWriter( outputStream, StandardCharsets.UTF_8 ) );
-    static final TimeZone UTC = TimeZone.getTimeZone( "UTC" );
-    public static final DateFormat SIMPLE_DATE_FORMAT = new SimpleDateFormat( "yyyy-MM-dd HH:mm:ss.SSSZ" );
 
     /**
      * A Builder for a {@link FormattedLog}
      */
     public static class Builder
     {
-        private TimeZone timezone = UTC;
+        private TimeZone timezone = FormattedLogger.UTC;
         private Object lock = this;
         private String category;
         private Level level = Level.INFO;
         private boolean autoFlush = true;
-        private DateFormat dateFormat = SIMPLE_DATE_FORMAT;
+        private Function<TimeZone, ZonedDateTime> dateTimeFunction = FormattedLogger.DEFAULT_CURRENT_DATE_TIME;
```

Comment:
Actually, I think it will be simpler to expose only following methods:

```
#withDateTimeFormatter(DateTimeFormatter formatter);
#withTimeSupplier(Supplier<TemporalAccessor> timeSupplier);
```

formatter can include `ZoneId` and do the conversion. Then `FormattedLogger#time()` can simply call:
`formatter.format(timeSupplier.get())` and no `with*TimeZone()` methods are needed.